roles/matrix-synapse/templates/synapse/homeserver.yaml.j2

```diff
@@ -2426,16 +2461,17 @@ opentracing:
 redis:
   # Uncomment the below to enable Redis support.
   #
-  #enabled: true
+  enabled: true
 
   # Optional host and port to use to connect to redis. Defaults to
   # localhost and 6379
   #
-  #host: localhost
+  host: matrix-redis
```

Comment:
It probably makes sense to introduce a `matrix_synapse_redis_host` variable in the `matrix-synapse` role, with a default value of `""`.

And maybe add a check in `roles/matrix-synapse/tasks/validate_config.yml` to ensure it's not empty if `matrix_synapse_workers_enabled`.

And then to have something like this in `group_vars/matrix_servers`:

```yaml
matrix_synapse_redis_host: "{{ 'matrix-redis' if matrix_redis_enabled else '' }}"
```

This way, people could disable the integrated Redis (`matrix_redis_enabled: false`) and point `matrix_synapse_redis_host` to some other (potentially external host) if they wish to do so.

Well, I guess we may need configuration options for the Redis port and potentially password as well.

This is not very crucial and it's something which can be changed later on as well, so it doesn't hurt to skip it if you're busy with other things.